mixer/adapter/svcctrl/expiringLRUCache.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package svcctrl
+
+import (
+	"sync"
+	"time"
+
+	"github.com/hashicorp/golang-lru/simplelru"
+	"github.com/jonboulle/clockwork"
+)
+
+type (
+	cacheValue struct {
+		// insert timestamp
+		timestamp time.Time
+		value     interface{}
+	}
+
+	// A thread safe LRU cache with expiration.
+	expiringLRUCache struct {
+		mutex sync.RWMutex
+		lru   *simplelru.LRU
+		clock clockwork.Clock
```

Comment:
Mixer already has a lot of external dependencies we want to remove. I'm not happy about introducing yet another dependency. What does this clockwork component do for us?

Reply:
I didn't introduce this dependency. It is  in our WORKSPACE file. It provides a clock interface very useful for test.

Reply:
`clockwork` is a downstream dependency that was introduced by a direct dependency on some of k8s client libs.  It isn't used in any Istio code directly, prior to this PR.

It might be a good idea to avoid it, or recreate the interfaces, in the off-chance that the k8s deps slim.